pkg/build/builder/sti.go

```diff
@@ -204,19 +215,42 @@ func (s *S2IBuilder) Build() error {
 		return err
 	}
 
+	// TODO: why in docker builds we defer removing the image, while here we don't?!
```

Comment:
probably a bug, I don't see any reason to keep the image cached on the node locally when the image is pushed to remote registry (other than a performance issue when running all-in-one ;-) @csrwng any ideas?